micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jetty/JettyConnectionMetrics.java

```diff
@@ -0,0 +1,100 @@
+ * connectionMetrics.bindTo(registry);
+ *
+ * Server server = new Server(0);
+ * Connector connector = new ServerConnector(server);
+ * connector.addBean(serverConnectionStatistics);
+ * server.setConnectors(new Connector[] { connector });
+ * }</pre>
+ *
+ * @author Tom Akehurst
+ *
+ */
+public class JettyConnectionMetrics implements MeterBinder {
+
+    private final ConnectionStatistics connectionStatistics;
+    private final Iterable<Tag> tags;
+
+    public JettyConnectionMetrics(ConnectionStatistics connectionStatistics) {
+        this(connectionStatistics, emptyList());
+    }
+
+    public JettyConnectionMetrics(ConnectionStatistics connectionStatistics, Iterable<Tag> tags) {
+        this.connectionStatistics = connectionStatistics;
+        this.tags = tags;
+    }
+
+    @Override
+    public void bindTo(MeterRegistry registry) {
+        Gauge.builder("jetty.connector.connections.current", connectionStatistics, ConnectionStatistics::getConnections)
+                .tags(tags)
+                .description("The current number of open connections")
+                .register(registry);
+        Gauge.builder("jetty.connector.connections.max", connectionStatistics, ConnectionStatistics::getConnectionsMax)
+                .tags(tags)
+                .description("The maximum number of connections")
+                .register(registry);
+        Gauge.builder("jetty.connector.connections.total", connectionStatistics, ConnectionStatistics::getConnectionsTotal)
+                .tags(tags)
+                .description("The total number of connections")
+                .register(registry);
+
+        FunctionCounter.builder("jetty.connector.bytesReceived", connectionStatistics, ConnectionStatistics::getReceivedBytesRate)
+                .tags(tags)
+                .description("The rate of bytes received")
+                .baseUnit("bytes")
+                .register(registry);
+        FunctionCounter.builder("jetty.connector.bytesSent", connectionStatistics, ConnectionStatistics::getSentBytesRate)
+                .tags(tags)
+                .description("The rate of bytes sent")
+                .baseUnit("bytes")
+                .register(registry);
+
+        TimeGauge.builder("jetty.connector.duration.max", connectionStatistics, MILLISECONDS, ConnectionStatistics::getConnectionDurationMax)
+                .tags(tags)
+                .description("The maximum connection lifetime duration")
+                .register(registry);
+        TimeGauge.builder("jetty.connector.duration.mean", connectionStatistics, MILLISECONDS, ConnectionStatistics::getConnectionDurationMean)
+                .tags(tags)
+                .description("The mean connection lifetime duration")
+                .register(registry);
+        TimeGauge.builder("jetty.connector.duration.stddev", connectionStatistics, MILLISECONDS, ConnectionStatistics::getConnectionDurationStdDev)
```

Comment:
I'm still hoping to get some feedback on whether including mean and standard deviation for duration is even useful. After we release this in a version, it's hard to remove the metrics because it will break i.e. metrics dashboards for users. Maybe it would be better for us to properly instrument this with Micrometer ourselves, rather than only relying on what is available via `ConnectionStatistics`. Or we could request an enhancement in Jetty for better duration statistics.

Reply:
I'd have thought mean would be the most useful of the three available metrics, although I'll admit I'm not wholly convinced there's much utility in any of them. I'm struggling to think what tuning decisions I might make based on connection duration.

Perhaps @john-karp can comment?

Reply:
On mean latency, here's a [short blog post](http://latencytipoftheday.blogspot.com/2014/06/latencytipoftheday-average-random.html) with the summary of arguments against using it.

Reply:
Oh yeah, I get that mean is a poor way to present latency. I'm not sure connection lifetime can be considered a latency metric though.

Reply:
Maybe I misunderstand what the connection duration means here. Is it the duration the server connection is open? Isn't that essentially the server response time to requests? Or is the duration across more than one request?

Reply:
Connections are usually reused, so in most cases it'll be the lifetime of the connection across many request/responses.

Reply:
Our use case is that we're using HTTP as a transport for batches of data. We want to detect if incoming connections are short-lived, since frequent TLS connection (re)establishment reduces our throughput. I agree the standard deviation and max as computed by Jetty are not that useful.

Reply:
In the past I've solved this problem by adding a rate metric derived from total connections. Although it's not quite the same as tracking connection lifetime, when compared with the request rate, it gives a good indication of whether connections are being well utilised.

I guess you could even explicitly calculate a ratio of the two in your monitoring tool.

Reply:
OK, how's this for a final set of changes?

* Remove the connection duration metrics completely for now
* Rename the rate metrics to remove "bytes" from the names, and add the unit

Reply:
Ok, that's fine. I have one other comment though.